app/controllers/open_access_publications_controller.rb

```diff
@@ -57,6 +69,48 @@ def scholarsphere_file_version
     render :edit
   end
 
+  def file_version_result
+    job_ids = params[:job_ids]
+    pdf_file_versions = []
+    exif_file_versions = []
+    @cache_files = []
+    @file_version = nil
+
+    # Remove failed Delayed::Job and log an error
+    job_ids&.reject! do |job_id|
+      if Delayed::Job.exists?(job_id)
+        job = Delayed::Job.find(job_id)
+        if job.failed_at.nil?
+          false
+        else
+          job.destroy
+          Rails.logger.error "Job with ID #{job_id} failed and has been removed"
+          true
+        end
+      else
+        false
+      end
+    end
```

Comment:
I don't know how important it is that this bit happens, but if I'm understanding everything correctly, then it seems like this is dependent on the client polling for results of the file version check. So in other words, if the client doesn't poll (i.e. the user closes their browser or navigates away from the page), then this failed job cleanup and error logging doesn't occur. Is that correct? If so, would that be a problem?

Reply:
I added this part while writing the specs, but you are right, it seems that I made a wrong assumption. I will be revising this section. Thank you for bringing it to my attention.

Reply:
To answer your question about the cases if user closes browser and logging does not happen, that would not be a problem esp since it is not saving anything to database.